app/models/concerns/orchestration/dhcp.rb

```diff
@@ -79,7 +79,11 @@ def boot_server
     return unless tftp?
 
     # first try to ask our IPv4 TFTP server for its boot server
-    bs = tftp.bootServer
+    bs = subnet.tftp.setting('TFTP', 'tftp_servername')
+    if bs.nil? && tftp.bootServer
+      bs = tftp.bootServer
+      Foreman::Deprecation.deprecation_warning('1.23', 'Tftp server name assumption will be removed, please update your Smart Proxy to 1.22 or later.')
+    end
```

Comment:
First, the bootServer is also called from Proxy status page, we would need to copy the same code there. I have an attempt actually to move the logic from orchestration to the ProxyAPI layer: https://github.com/theforeman/foreman/pull/6465/files

Second, wasn't the purpose of exposing settings to actually drop the `bootServer` call completely? I don't understand the change, the call simply returns the setting. Now we have two endpoints providing the same thing twice - we should use one or another I guess.

Reply:
We can't assume all proxies are upgraded to the new version directly. That's why we deprecate it first. Given this will land in 1.22 (earliest) should we deprecate it for 1.24? @tbrisker?

Reply:
I think making it deprecated for 1.24 makes sense.

Reply:
updated to 1.24

Reply:
I am puzzled by "Tftp server name assumption will be removed", what exactly are we removing in 1.24? Can you elaborate and reformulate the statement to be more clear? I'd expect something like "Cannot read tftp_servername from proxy XYZ, please upgrade it".

Reply:
clarified message